### PR TITLE
Add Iterant.takeEveryNth operation

### LINKED ISSUE #500: Add Iterant.takeEveryNth
Need a new `Iterant` op that that emits every n-th event from the source, dropping intermediary events.

```scala
sealed abstract class Iterant[F[_], A] {
  // ...
  def takeEveryNth(n: Int)(implicit F: Sync[F]): Iterant[F, A]
}
```

See the [observable implementation](https://github.com/monix/monix/blob/v3.0.0-M3/monix-reactive/shared/src/main/scala/monix/reactive/internal/operators/TakeEveryNthOperator.scala) for inspiration.

### PR DESCRIPTION
**Pull request by greenhat**
Closes #500

### CODE REVIEW

File: monix-tail/shared/src/test/scala/monix/tail/IterantTakeEveryNthSuite.scala [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    val stop = Coeval.eval(effect += 1)
+    val source = Iterant[Coeval].nextCursorS(BatchCursor(1,2,3), Coeval.now(Iterant[Coeval].empty[Int]), stop)
+    val stream = source.takeEveryNth(1)
+    stream.earlyStop.value
+    assertEquals(effect, 1)
+  }
+
+  test("Iterant.takeEveryNth throws on invalid n") { implicit s =>
+    val source = Iterant[Coeval].nextCursorS(BatchCursor(1,2,3), Coeval.now(Iterant[Coeval].empty[Int]), Coeval.unit)
+    intercept[IllegalArgumentException] { source.takeEveryNth(0).toListL.value }
```
Comment (alexandru) [inline monix-tail/shared/src/test/scala/monix/tail/IterantTakeEveryNthSuite.scala]: Just a note — we need to be precise about what throws the error. Given you have a `require` in `takeEveryNth`, then you should delete `.toListL.value`.

File: monix-tail/shared/src/test/scala/monix/tail/IterantTakeEveryNthSuite.scala [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+  }
+
+  test("Iterant.takeEveryNth triggers early stop on exception") { _ =>
+    check1 { (iter: Iterant[Coeval, Int]) =>
+      val cancelable = BooleanCancelable()
+      val dummy = DummyException("dummy")
+      val suffix = Iterant[Coeval].nextCursorS[Int](new ThrowExceptionCursor(dummy), Coeval.now(Iterant[Coeval].empty), Coeval.unit)
+      val stream = (iter.onErrorIgnore ++ suffix).doOnEarlyStop(Coeval.eval(cancelable.cancel()))
+      intercept[DummyException] { stream.takeEveryNth(1).toListL.value }
```
Comment (alexandru) [inline monix-tail/shared/src/test/scala/monix/tail/IterantTakeEveryNthSuite.scala]: Here you need to be precise about how the error gets thrown too. So the test should be:

```scala
assertEquals(stream.takeEveryNth(1).toListL.runTry, Failure(dummy))
```

**Comment by greenhat** [comment]
@alexandru Thanks! Fixed the places where the exception is expected. 
I'm done with implementation and I'm out of ideas for more tests. Let me know if I miss something.

File: monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+private[tail] object IterantTakeEveryNth {
+  /**
+    * Implementation for `Iterant#takeEveryNth`
+    */
+  def apply[F[_], A](source: Iterant[F, A], n: Int)
+                    (implicit F: Sync[F]): Iterant[F, A] = {
+
+    require(n > 0, "n must be strictly positive")
+
+    if (n == 1) return source
```
Comment (alexandru) [inline monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala]: I don't like usage of `return`, unless it's made for concrete optimization reasons, or in other words it's very low level stuff.

This check should be included before the `source match` below.

Comment (greenhat) [reply monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala]: @alexandru Totally agree. Just could not find a better place for it. Fixed.

File: monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+import monix.tail.internal.IterantUtils._
+
+import scala.collection.mutable.ArrayBuffer
+
+private[tail] object IterantTakeEveryNth {
+  /**
+    * Implementation for `Iterant#takeEveryNth`
+    */
+  def apply[F[_], A](source: Iterant[F, A], n: Int)
+                    (implicit F: Sync[F]): Iterant[F, A] = {
```
Comment (alexandru) [inline monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala]: Don't like this indentation style and the Monix code is pretty consistent on not doing things dangling to the right like that 😀 personal preference, but it's good to be consistent.

Put this signature all on one line, delete the empty line between it and `require` below and you're done 😀

**Comment by alexandru** [comment]
This looks good @greenhat, thanks, one last comment about indentation and you're done.

Will then merge as soon as the build is green.

File: monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala (hunk 4 above)
Comment (greenhat) [reply monix-tail/shared/src/main/scala/monix/tail/internal/IterantTakeEveryNth.scala]: Consistency is the king! Fixed. It's my first PR here, need to learn codebase more. Thanks for the fast and thorough review!

**Pull request merged**